src/components/Modal/BaseModal.js

```diff
@@ -30,17 +30,28 @@ class BaseModal extends PureComponent {
         this.hideModal = this.hideModal.bind(this);
     }
 
+    componentDidMount() {
+        if (!this.props.isVisible) { return; }
+
+        // to handle case of modal already visible when mounted, i.e. PopoverReportActionContextMenu
+        Modal.setCloseModal(this.props.onClose);
+    }
+
     componentDidUpdate(prevProps) {
         if (prevProps.isVisible === this.props.isVisible) {
             return;
         }
 
         Modal.willAlertModalBecomeVisible(this.props.isVisible);
+        Modal.setCloseModal(this.props.isVisible ? this.props.onClose : null);
     }
 
     componentWillUnmount() {
         // we don't want to call the onModalHide on unmount
         this.hideModal(this.props.isVisible);
+
+        // to handle case of modal unmounted with visible state
```

Comment:
Can we explain why this is needed with the comment. Ideally the comment should explain why we're doing something when it's non obvious. (the comment on line 50 is not a good example haha)

Reply:
```suggestion
        // To prevent closing any modal already unmounted when this modal still remains as visible state
```
@puneetlath how about this?

Reply:
Much better!